### PR TITLE
fix channel header is broken when opening channel with chat history after joining the community

### LINKED ISSUE #19671: Channel header is broken when opening channel with chat history after joining the community
Example of community https://status.app/c/iw6AChwKCE5ldyBjb21tEgVkYXNzZBgXIgcjNDM2MERGAw==#zQ3shibjPWsBFi527PL5hyBWVq9tWjuXdSXwBscTKgEEx3YwM

Steps: 

1. Join given community
2. Wait a little bit before community is fully fetched (5-10 seconds)
3. Open general channel
4. Pay attention at header

**Actual result:** Header is broken. Message history is overlapping status bar. 

photo_2024-04-16 16 19 37

https://github.com/status-im/status-mobile/assets/97245802/765cd920-eb47-44e5-8802-3895abf21cb0

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19671

status: ready

### CODE REVIEW

File: src/status_im/contexts/chat/messenger/messages/navigation/view.cljs [hunk 1]
```diff
@@ -132,12 +132,11 @@
                    [all-loaded-sub])
     [rn/view
      {:style (style/navigation-view navigation-view-height messages.constants/pinned-banner-height)}
-     (when (seq last-message)
```
Comment (Parveshdhull) [inline src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: don't know why we added this

Comment (ibrkhalil) [reply src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: From what I can remember, It was a fix for [this](https://github.com/status-im/status-mobile/pull/19459#issuecomment-2044480588)

Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: thank you @ibrkhalil for adding context. I am still confused, how last-message relates to shadow :thinking:

Comment (ibrkhalil) [reply src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: Sorry for not being specific, It's the overlapping of the top nav bar with the avatar.

Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: Thank you. Unfortunately, I am not able to reproduce this overlap, if this ever comes up let's log an issue perhaps with a video to give more context.

Also, I don't think the last message is a proper fix if the top bar overlaps. We should not just remove the top bar when there is no last message. If it overlaps then probably something wrong with the animation, as you can see from the shared image position of an avatar with respect to the close button is wrong.


image

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im/contexts/chat/messenger/messages/navigation/view.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im/contexts/chat/messenger/messages/navigation/view.cljs:140]: I am requesting design review as it was approved on my PR.

**Comment by pavloburykh** [comment]
@Parveshdhull thank you for the fix! @Francesca-G could you please review everything related to channel header in this PR? Thank you!

**Review by Francesca-G — COMMENTED** [review]
The animation when scrolling looks good, but we have a couple of weird states when the channel is empty:

- This happens when opening the composer:

CEDE4298-FC37-4284-BDA4-8C83C5C46F17_1_102_o


- This happens when there's enough messages to get the animation "started" but not quite enough to have the full animation:

5E0B8309-3803-4656-8F27-05BAF5A6C0B5_1_102_o

**Comment by pavloburykh** [comment]
> The animation when scrolling looks good, but we have a couple of weird states when the channel is empty:

> - This happens when opening the composer:

> CEDE4298-FC37-4284-BDA4-8C83C5C46F17_1_102_o

Thanks @Francesca-G! This is known and fixed in another PR https://github.com/status-im/status-mobile/pull/19679. I am moving PR to merge if there no other issues.

**Comment by pavloburykh** [comment]
I am not sure about this one. Probably not related to PR and can be fixed separately, WDYT @Parveshdhull? 

> This happens when there's enough messages to get the animation "started" but not quite enough to have the full animation 

5E0B8309-3803-4656-8F27-05BAF5A6C0B5_1_102_o

**Comment by Parveshdhull** [comment]
Thank you @Francesca-G and @pavloburykh for testing the PR.

> This happens when there's enough messages to get the animation "started" but not quite enough to have the full animation

Yes this is known issue (Chat animation gets stuck mid way) and logged here https://github.com/status-im/status-mobile/issues/19639

> This happens when opening the composer

This is also related to top-bar animation. If you checkout in the shared image, avatar is overlapping topbar because position of avatar is changed without changing its size etc. I think this one also should be fixed while addressing other top-bar issues.

Will it be ok if I log this one with https://github.com/status-im/status-mobile/issues/19639

**Comment by Parveshdhull** [comment]
hi @pavloburykh, I logged the topbar overlap issue in https://github.com/status-im/status-mobile/issues/19639. Is PR ready for merging?

**Comment by pavloburykh** [comment]
> hi @pavloburykh, I logged the topbar overlap issue in #19639. Is PR ready for merging?

Yes @Parveshdhull, ready for merge. Thank you!

**Pull request merged**